src/nilrag/util.py

```diff
@@ -102,6 +103,33 @@ def find_closest_chunks(
     sorted_indices = np.argsort(distances)
     return [(chunks[idx], distances[idx]) for idx in sorted_indices[:top_k]]
 
+def get_closest_centroid(prompt: str, centroids: list[int]) -> list[int]:
```

Comment:
Can we pass the query_embedding here instead of a string? I think `generate_embeddings_huggingface` shouldn't be part of this function